.ci/test_r_package_windows.ps1

```diff
@@ -12,32 +12,53 @@ function Download-File-With-Retries {
   } while(!$?);
 }
 
-$env:R_WINDOWS_VERSION = "3.6.3"
 $env:R_LIB_PATH = "$env:BUILD_SOURCESDIRECTORY/RLibrary" -replace '[\\]', '/'
 $env:R_LIBS = "$env:R_LIB_PATH"
-$env:PATH = "$env:R_LIB_PATH/Rtools/bin;" + "$env:R_LIB_PATH/R/bin/x64;" + "$env:R_LIB_PATH/miktex/texmfs/install/miktex/bin/x64;" + $env:PATH
+$env:PATH = "$env:R_LIB_PATH/Rtools/bin;" + "$env:R_LIB_PATH/Rtools/usr/bin;" + "$env:R_LIB_PATH/R/bin/x64;" + "$env:R_LIB_PATH/miktex/texmfs/install/miktex/bin/x64;" + $env:PATH
 $env:CRAN_MIRROR = "https://cloud.r-project.org/"
 $env:CTAN_MIRROR = "https://ctan.math.illinois.edu/systems/win32/miktex/tm/packages/"
 
-if ($env:COMPILER -eq "MINGW") {
-  $env:CXX = "$env:R_LIB_PATH/Rtools/mingw_64/bin/g++.exe"
-  $env:CC = "$env:R_LIB_PATH/Rtools/mingw_64/bin/gcc.exe"
+# Get details needed for installing R components
+#
+# NOTES:
+#    * some paths and file names are different on R4.0
+$env:R_MAJOR_VERSION=$env:R_VERSION.split('.')[0]
+if ($env:R_MAJOR_VERSION -eq "3") {
+  $env:RTOOLS_MINGW_BIN = "$env:R_LIB_PATH/Rtools/mingw_64/bin"
+  $env:RTOOLS_EXE_FILE = "Rtools35.exe"
+  $env:R_WINDOWS_VERSION = "3.6.3"
+} elseif ($env:R_MAJOR_VERSION -eq "4") {
+  $env:RTOOLS_MINGW_BIN = "$env:R_LIB_PATH/Rtools/mingw64/bin"
+  $env:RTOOLS_EXE_FILE = "rtools40-x86_64.exe"
+  $env:R_WINDOWS_VERSION = "4.0.0"
+} else {
+  Write-Output "Unrecognized R version: $env:R_VERSION"
+  Check-Output $false
+}
+
+if ($env:COMPILER -ne "MINGW") {
```

Comment:
I'm confused here. Maybe it should be `eq`?

Reply:
yes you're right, that was a mistake

Reply:
this is getting confusing because we now have two dimensions (`COMPILER` and `TOOLCHAIN`). Soon, if we follow the proposal I'll put up for CRAN (https://github.com/jameslamb/LightGBM/pull/25), we'll have a third dimension: `R_BUILD_TYPE` which is either `cmake` or `CRAN`.

I think it might make sense to refactor this file (in a later PR) into some functions like:

* `Install-R`
* `Install-Miktex-And-Pandoc`
* `Run-R-CMD-Check`

Then the top half of the file would define the logic, and the bottom half would look more like ` .travis.yml` build matrix, where you can tell which pieces run for each combination. Something like this pseudocode:

```powershell
Install-R-and-Rtools
if ($env:COMPILER -eq "MSVC") {
    Build-With-Cmake -SkipInstall $false
    Run-Tests-With-Testthat
} else {
   Install-Miktex-and-Pandoc 
   if ($env:TOOLCHAIN -eq "MINGW") {
        Build-With-Cmake -SkipInstall $true -Use "MINGW"
    } elseif ($env:TOOLCAHIN -eq "MSYS") {
        Build-With-Cmake -SkipInstall $true -Use "MSYS"
   }
   Run-R-CMD-Check
}
```

Reply:
> we'll have a third dimension: R_BUILD_TYPE which is either cmake or CRAN.

Oh my! 😄 

> I think it might make sense to refactor this file (in a later PR) into some functions like:

👍